src/Lean/Data/Format.lean

```diff
@@ -73,16 +73,16 @@ def spaceUptoLine' : List (Nat × Format) → Nat → SpaceResult
 | [],    w => {}
 | p::ps, w => merge w (spaceUptoLine p.2 w) (spaceUptoLine' ps w)
 
-partial def be : Nat → Nat → String → List (Nat × Format) → String
-| w, k, out, []                        => out
-| w, k, out, (i, nil)::z               => be w k out z
-| w, k, out, (i, (compose _ f₁ f₂))::z => be w k out ((i, f₁)::(i, f₂)::z)
-| w, k, out, (i, (nest n f))::z        => be w k out ((i+n, f)::z)
-| w, k, out, (i, text s)::z            => be w (k + s.length) (out ++ s) z
-| w, k, out, (i, line)::z              => be w i ((out ++ "\n").pushn ' ' i) z
-| w, k, out, (i, choice f₁ f₂)::z      =>
+partial def be : Nat → String → List (Nat × Format) → String
+| w, out, []                        => out
+| w, out, (i, nil)::z               => be w out z
+| w, out, (i, (compose _ f₁ f₂))::z => be w out ((i, f₁)::(i, f₂)::z)
+| w, out, (i, (nest n f))::z        => be w out ((i+n, f)::z)
+| w, out, (i, text s)::z            => be w (out ++ s) z
+| w, out, (i, line)::z              => be w ((out ++ "\n").pushn ' ' i) z
+| w, out, (i, choice f₁ f₂)::z      =>
   let r := merge w (spaceUptoLine f₁ w) (spaceUptoLine' z w);
```

Comment:
Scratch that, `k` never being used is actually a bug: `k` represents the line length so far and we should use `w-k` as the remaining space. In Wadler's implementation this is `if fits (w-k) x then x else y`. Will fix.